doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,613 @@
+    - [Phase III](#phase-iii)
+- [2 Architecture Design](#2-architecture-design)
+- [3 Modules Design](#3-modules-design)
+  - [3.1 Config DB](#31-config-db)
+    - [3.1.1 MACsec Profile Table](#311-macsec-profile-table)
+    - [3.1.2 Port Table](#312-port-table)
+  - [3.2 App DB](#32-app-db)
+    - [3.2.1 MACsec Port Table](#321-macsec-port-table)
+    - [3.2.2 MACsec Egress SC Table](#322-macsec-egress-sc-table)
+    - [3.2.3 MACsec Ingress SC Table](#323-macsec-ingress-sc-table)
+    - [3.2.4 MACsec Egress SA Table](#324-macsec-egress-sa-table)
+    - [3.2.5 MACsec Ingress SA Table](#325-macsec-ingress-sa-table)
+  - [3.3 State DB](#33-state-db)
+    - [3.3.1 MACsec Port Table](#331-macsec-port-table)
+    - [3.3.2 MACsec Egress SC Table](#332-macsec-egress-sc-table)
+    - [3.3.3 MACsec Ingress SC Table](#333-macsec-ingress-sc-table)
+    - [3.3.4 MACsec Egress SA Table](#334-macsec-egress-sa-table)
+    - [3.3.5 MACsec Ingress SA Table](#335-macsec-ingress-sa-table)
+  - [3.4 Software Modules](#34-software-modules)
+    - [3.4.1 MACsec Mgr](#341-macsec-mgr)
+      - [3.4.1.1 Primary/Fallback decision](#3411-primaryfallback-decision)
+      - [3.4.1.2 wpa_cli parameters list](#3412-wpa_cli-parameters-list)
+      - [3.4.1.3 Enable MACsec](#3413-enable-macsec)
+      - [3.4.1.4 Disable MACsec](#3414-disable-macsec)
+    - [3.4.2 WPA Supplicant](#342-wpa-supplicant)
+      - [3.4.2.1 Extension Parameters](#3421-extension-parameters)
+      - [3.4.2.2 Proactive SAK refresh](#3422-proactive-sak-refresh)
+      - [3.4.2.3 Scalability Evaluation](#3423-scalability-evaluation)
+    - [3.4.3 SONiC MACsec Plugin](#343-sonic-macsec-plugin)
+    - [3.4.4 MACsec Orch](#344-macsec-orch)
+      - [3.4.4.1 Functions](#3441-functions)
+      - [3.4.4.2 Flex Counter Interval](#3442-flex-counter-interval)
+    - [3.4.5 vMACsec SAI](#345-vmacsec-sai)
+- [4 Flow](#4-flow)
+  - [4.1 Init Port](#41-init-port)
+  - [4.2 Init MACsec](#42-init-macsec)
+  - [4.3 Create and Enable SA](#43-create-and-enable-sa)
+
+## About this Manual
+
+This document provides general information about the MACsec feature implementation in SONiC.
+
+## Abbreviation
+
+| Abbreviation | Description                              |
+| ------------ | ---------------------------------------- |
+| CA           | Secure Connectivity Association          |
+| CAK          | Secure Connectivity Association Key      |
+| CAN          | Secure Connectivity Association Key Name |
+| SA           | Secure Association                       |
+| SAK          | Secure Association Key                   |
+| SC           | Secure Channel                           |
+| SCI          | Secure Channel Identifier                |
+
+## 1 Requirements Overview
+
+### 1.1 Functional requirements
+
+This section describes the SONiC requirements for MACsec primary feature.
+
+At a high level the following should be supported:
+
+#### Phase I
+
+- MACsec should be supported on physical port
```

Comment:
May be we should also provide a reference to PortChannels

Reply:
Done, please review it.